src/components/BookDetails/BookDetails.tsx

```diff
@@ -0,0 +1,49 @@
+// src/components/BookDetails/BookDetails.tsx
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { BookType } from "../BookList/BookList";
+import "./BookDetails.scss";
+
+interface Params {
+  id: string;
+  [key: string]: string | undefined;
+}
+
+export const BookDetails: React.FC = () => {
+  const { id } = useParams<Params>();
+  const [book, setBook] = useState<BookType | null>(null);
+
+  useEffect(() => {
```

Comment:
please move this fetch funcion above useEffect

Reply:
Fixed